include/stout/flags/flags.cc

```diff
@@ -587,18 +569,39 @@ void Parser::Parse(const std::vector<ArgumentInfo>& values) {
   }
 
   // Ensure required flags are present.
-  for (const auto& [_, field] : fields_) {
+  for (const auto& [name, field] : fields_) {
+    // The field has been parsed, all good, just continue.
+    if (parsed_.count(field) > 0) {
+      continue;
+    }
     const stout::v1::Flag& flag =
         field->options().GetExtension(stout::v1::flag);
 
-    // This field is not required, so just continue.
-    if (!flag.required())
+    // Need to normalize string default value before parsing.
+    // NOTE: for some reason protoc compiler has generated
+    // 'default_()' method with underscore, we believe that
+    // this happens because 'default' conflicts with reserved
+    // words.
+    const std::string normalized_value =
+        GetNormalizedDefaultValue(
+            flag.default_(),
+            field->type() == google::protobuf::FieldDescriptor::TYPE_STRING);
+
+    if (flag.has_default_() && !flag.required()) {
+      SetFieldMessageOrAggregateErrors(
+          normalized_value,
+          name,
+          field,
+          &GetMessageForField(*field),
+          errors);
       continue;
+    }
 
-    // Required field has been parsed, so it is present.
-    // All good, just continue.
-    if (parsed_.count(field) > 0)
+    // This field is not required, and there is no any default value
```

Comment:
I'd find this less confusing if we moved this `!required` if statement up above the `has_default_()` if statement above.

That way we'd skip all required fields (since none of them can have defaults per our new validation), then afterwards we'd just check for defaults (and we wouldn't need to re-check required, since we already checked that in the previous if statement)